File: LeetCodeProblemsCMake/Problems/LeetCode2894DivisibleAndNondivisibleSumsDifference.cpp
//
// Created by saeed on 5/27/25.
//

#include "LeetCode2894DivisibleAndNondivisibleSumsDifference.h"

int LeetCode2894DivisibleAndNondivisibleSumsDifference::differenceOfSums(int n, int m) {
    long long sum=(1+n);
    sum*=n;
    sum/=2;
    long long divisable=m+((n/m)*m);
    divisable*= (n/m);
    divisable/=2;
    return sum-2*divisable;
}

File: LeetCodeProblemsGo/LeetCode2894DivisibleAndNondivisibleSumsDifference/LeetCode2894DivisibleAndNondivisibleSumsDifference.go
package main

func differenceOfSums(n int, m int) int {
	var sum int64 = int64(1 + n)
	sum *= int64(n)
	sum /= 2
	var sumDivisible int64 = int64(m + (n/m)*m)
	sumDivisible *= int64(n / m)
	sumDivisible /= 2
	return int(sum - 2*sumDivisible)
}
